File: model/src/main/java/com/blackopalsolutions/goneviral/model/response/GetCardsResponse.java
package com.blackopalsolutions.goneviral.model.response;

import com.blackopalsolutions.goneviral.model.domain.Card;

import java.util.List;

public class GetCardsResponse extends Response {
    private final List<Card> cards;

    public GetCardsResponse(List<Card> cards) {
        super(true);
        this.cards = cards;
    }

    public GetCardsResponse(String message) {
        super(message);
        this.cards = null;
    }

    public List<Card> getCards() {
        return cards;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/response/GetCardsResponse.kt
package com.blackopalsolutions.goneviral.model.response

import com.blackopalsolutions.goneviral.model.domain.Card

class GetCardsResponse : Response {
    val cards: List<Card>?

    constructor(cards: List<Card>) : super(true) {
        this.cards = cards
    }

    constructor(message: String) : super(message) {
        this.cards = null
    }
}